django_sirene/importers.py

```diff
@@ -126,172 +139,192 @@ def _prepare_relateds(self, params, row):
+        logger.info("%s institutions updated", len(self.to_update))
+        self.to_update = []
 
-    def _run(self, headquarters_import):
+    def _run_row(self, index, row):
+        # Filter by date to lighten the import
+        try:
+            last_update = datetime.strptime(
+                row["dateDernierTraitementEtablissement"], "%d/%m/%Y"
+            )
+            is_fresh = last_update and last_update >= self.date_from
+        except ValueError:
+            is_fresh = False
 
-        self._preload_data()
+        if not is_fresh and not self.force:
+            return
 
-        start = time.time()
-        for i, row in enumerate(self.rows, 1):
-            if headquarters_import != self._is_headquarter(row):
-                # we don't import headquarters and subsidiaries in same time
-                continue
-
-            vmaj = row.get('VMAJ', '').upper()
-            if vmaj == VMAJ.edit_initial:
-                # ignore this line because it's the initial state we want only
-                # the finale state where VMAJ = 'F'
-                continue
-
-            current_siret = row['SIREN'] + row['NIC']
-
-            already_exists = current_siret in self.db_all_sirets
-
-            params = self._prepare_institution_params(row)
-
-            if not self._is_headquarter(row):
-                # get headquarter, sometimes headquarter is removed
-                # but not subsidiaries
-                pk_headquarter = self.db_headquarters.get(row['SIREN'])
-                if pk_headquarter:
-                    params.update({
-                        'headquarter_id': pk_headquarter,
-                    })
-
-            self._prepare_relateds(params, row)
-            new_institution = Institution(**params)
-            if already_exists:
-                self.to_update.append(new_institution)
-            else:
-                self.to_create.append(new_institution)
+        already_exists = row["siret"] in self.db_all_sirets
+        params = self._prepare_institution_params(row)
+        self._prepare_relateds(params, row)
 
-            # treat in batch to control memory consumption
-            if len(self.to_create) >= self.local_batch_size:
-                self._create_in_db()
-            if len(self.to_update) >= self.local_batch_size:
-                self._update_db()
+        new_institution = Institution(**params)
+        if already_exists:
+            self.to_update.append(new_institution)
+        else:
+            self.to_create.append(new_institution)
 
-            # make some log
-            if self.log and i % self.local_batch_size == 0:
-                end = time.time()
-                items_by_sec = self.local_batch_size / (end - start)
-                logger.debug(
-                    'Treated {} rows ({:0.0f} items/sec)'.format(i, items_by_sec),
-                )
-                start = time.time()
+    def run(self):
+        super().run()
```

Comment:
I think you don't need that.

Reply:
Yes I do otherwise it won't be called :wink:

Reply:
I didn't see the following lines, my bad!